src/gluonts/model/deepar/_schema.py

```diff
@@ -0,0 +1,155 @@
+        if self.ndim is not None and self.ndim != value.ndim:
+            raise GluonTSDataError(
+                f"expected array with dimension {self.ndim}, "
+                "but got {value.ndim}."
+            )
+        return value
+
+    def is_compatible(self, value: Any) -> bool:
+        if not isinstance(value, (list, tuple, np.ndarray)):
+            return False
+
+        try:
+            x = np.asarray(value)
+        except Exception:
+            return False
+
+        # int types
+        if self.dtype in [np.int32, np.int64]:
+            if x.dtype.kind != "i":
+                return False
+            return self.ndim is None or x.ndim == self.ndim
+
+        # float types
+        try:
+            x = np.asarray(value, dtype=self.dtype)
+        except Exception:
+            return False
+        return self.ndim is None or x.ndim == self.ndim
+
+
+Freq = Union[str, pd.DateOffset]
+
+
+class PandasPeriodField(FieldType[pd.Period]):
+    def __init__(
+        self,
+        freq: Freq,
+    ) -> None:
+        self.freq = to_offset(freq) if isinstance(freq, str) else freq
+
+    def __eq__(self, other):
+        if not isinstance(other, PandasPeriodField):
+            return False
+        return self.freq == other.freq
+
+    def __repr__(self):
+        return f"PandasPeriodField(freq={self.freq!r})"
+
+    def __call__(self, value: Any) -> pd.Period:
+        return pd.Period(value, freq=self.freq)
+
+    def is_compatible(self, v: Any) -> bool:
+        if not isinstance(v, (str, pd.Period)):
+            return False
+        try:
+            self(v)
+        except Exception:
+            return False
+        return True
+
+
+@dataclass
+class Schema:
+    fields: Dict[str, FieldType]
+    default_values: Dict[str, Any]
+
+    def __call__(
+        self, d: Dict[str, Any], inplace: bool
+    ) -> Dict[str, FieldType]:
+        """
+        inplace
+            True: applies the schema to the input dictionary.
+                  The dictionary is updated in place.
+            False: return a new data dictionary if False.
+        """
+        if inplace:
+            out: Dict[str, Any] = d
+        else:
+            out = {}
+        for field_name, field_type in self.fields.items():
+            try:
+                value = d[field_name]
+            except KeyError:
+                if field_name in self.default_values.keys():
+                    value = self.default_values[field_name]
+                else:
+                    raise GluonTSDataError(
+                        f"field {field_name} does not occur in the data"
+                    )
+            try:
+                out[field_name] = field_type(value)
+            except Exception as e:
+                raise GluonTSDataError(
+                    f"Error when processing field {field_name} using "
+                    "{field_type}"
+                ) from e
+        if inplace:
+            remove_field_names = []
```

Comment:
I'm not sure we want to do this, but I also see why we would want to.

Maybe have it as an option?

Reply:
Yes, `SelectFields` will do it later.
```
return TrainDataLoader(
            dataset=data,
            transform=instance_splitter + SelectFields(input_names),
            batch_size=self.batch_size,
            stack_fn=partial(batchify, ctx=self.trainer.ctx, dtype=self.dtype),
            decode_fn=partial(as_in_context, ctx=self.trainer.ctx),
            **kwargs,
        )
```
So I think I can remove it here.